src/header/common/link.rs

```diff
@@ -0,0 +1,975 @@
+
+    /// Reverse Relation Types: `rev`
+    rev: Option<Vec<RelationType>>,
+
+    /// Language Tags: `hreflang`
+    href_lang: Option<Vec<LanguageTag>>,
+
+    /// Media Descriptors: `media`
+    media_desc: Option<Vec<MediaDesc>>,
+
+    /// Quoted String: `title`
+    title: Option<String>,
+
+    /// Extended Value: `title*`
+    title_star: Option<String>,
+
+    /// Media Type: `type`
+    media_type: Option<Mime>,
+
+    /// Link Extension: `link-extensions`
+    link_extension: Option<String>
+}
+
+/// A Media Descriptors Enum based on
+/// https://www.w3.org/TR/html401/types.html#h-6.13
+#[derive(Clone, PartialEq, Debug)]
+pub enum MediaDesc {
+    /// screen
+    Screen,
+    /// tty
+    Tty,
+    /// tv
+    Tv,
+    /// projection
+    Projection,
+    /// handheld
+    Handheld,
+    /// print
+    Print,
+    /// braille
+    Braille,
+    /// aural
+    Aural,
+    /// all
+    All,
+    /// Other Values
+    Value(String)
+}
+
+/// A Link Relation Type Enum based on
+/// [RFC5988](https://tools.ietf.org/html/rfc5988#section-6.2.2)
+#[derive(Clone, PartialEq, Debug)]
+pub enum RelationType {
+    /// alternate
+    Alternate,
+    /// appendix
+    Appendix,
+    /// bookmark
+    Bookmark,
+    /// chapter
+    Chapter,
+    /// contents
+    Contents,
+    /// copyright
+    Copyright,
+    /// current
+    Current,
+    /// describedby
+    DescribedBy,
+    /// edit
+    Edit,
+    /// editMedia
+    EditMedia,
+    /// enclosure
+    Enclosure,
+    /// first
+    First,
+    /// glossary
+    Glossary,
+    /// help
+    Help,
+    /// hub
+    Hub,
+    /// index
+    Index,
+    /// last
+    Last,
+    /// latestVersion
+    LatestVersion,
+    /// license
+    License,
+    /// next
+    Next,
+    /// nextArchive
+    NextArchive,
+    /// payment
+    Payment,
+    /// prev
+    Prev,
+    /// predecessorVersion
+    PredecessorVersion,
+    /// previous
+    Previous,
+    /// prevArchive
+    PrevArchive,
+    /// related
+    Related,
+    /// replies
+    Replies,
+    /// section
+    Section,
+    /// self
+    RelationTypeSelf,
+    /// service
+    Service,
+    /// start
+    Start,
+    /// stylesheet
+    Stylesheet,
+    /// subsection
+    Subsection,
+    /// successorVersion
+    SuccessorVersion,
+    /// up
+    Up,
+    /// versionHistory
+    VersionHistory,
+    /// via
+    Via,
+    /// working-copy
+    WorkingCopy,
+    /// working-copy-of
+    WorkingCopyOf,
+    /// ext-rel-type
+    ExtRelType(Uri)
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Struct methods
+////////////////////////////////////////////////////////////////////////////////
+
+impl Link {
+    /// Create Link from a `Vec<LinkValue>`
+    pub fn new(link_values: Vec<LinkValue>) -> Link {
+        Link { values: link_values }
+    }
+}
+
+#[allow(dead_code)]
+impl LinkValue {
+    /// Create LinkValue from URI-Reference
+    pub fn new(uri: &str) -> ::Result<LinkValue> {
```

Comment:
Returning a `Result` from the constructor doesn't look very good. I did it like that because `uri::Uri` is not public. We could change that if you want to.